flatten.go

```diff
@@ -0,0 +1,159 @@
+package authzed
+
+import (
+	"fmt"
+
+	api "github.com/authzed/authzed-go/arrakisapi/api"
+)
+
+// Flatten reduces an ExpandResponse into the slice of Users present in the
+// expansion.
+func Flatten(resp *api.ExpandResponse) []*api.User { return flatten(resp.TreeNode) }
+
+func flatten(node *api.RelationTupleTreeNode) []*api.User {
+	switch typed := node.NodeType.(type) {
+	case *api.RelationTupleTreeNode_IntermediateNode:
+		switch typed.IntermediateNode.Operation {
+		case api.SetOperationUserset_UNION:
+			return flattenUnion(typed.IntermediateNode.ChildNodes)
+		case api.SetOperationUserset_INTERSECTION:
+			return flattenIntersection(typed.IntermediateNode.ChildNodes)
+		case api.SetOperationUserset_EXCLUSION:
+			return flattenExclusion(typed.IntermediateNode.ChildNodes)
+		}
+	case *api.RelationTupleTreeNode_LeafNode:
+		var toReturn userSet = make(map[string]struct{})
```

Comment:
`toReturn := newUserSet()` and then add a method

Reply:
do you think we really need this for something private?